tracks/perl5/two-fer/mentoring.md

```diff
@@ -0,0 +1,111 @@
+# Mentoring
+
+This is a great opportunity to prepare for the track with attention to style conventions.
+This is currently the first reviewed problem on the Perl track, and serves to
+make sure the student understands unit testing and Test Driven Design.
+
+## Reasonable Solutions
+
+```perl
+package TwoFer;
+use strict;
+use warnings;
+use Exporter 'import';
+our @EXPORT_OK = qw(two_fer);
+sub two_fer {
+    my $name = shift || "you";
+    return "One for $name, one for me.";
+}
+1;
+```
+
+```perl
+package TwoFer;
+use strict;
+use warnings;
+use Exporter 'import';
+our @EXPORT_OK = qw(two_fer);
+sub two_fer {
+    return "One for " . (shift || "you") . ", one for me.";
+}
+1;
+```
+
+```perl
+package TwoFer;
+use strict;
+use warnings;
+use Exporter 'import';
+our @EXPORT_OK = qw(two_fer);
+use experimental qw(signatures);
+sub two_fer ($name = 'you') {
+    return "One for $name, one for me.";
+}
+1;
+```
+But remember TMTOWDI - There's More Than One Way To Do It!
+
+## Common Suggestions
+
+- Suggest using a default value instead of any form of conditionals.
+- If the student is using Perl 5.2 or newer, suggest using the `experimentals` module but make them aware that they're unlikely to find that particular pattern in the wild.
+- If the student is using Perl older than 5.2, suggest upgrading to get new features - but, again, make them aware that not every situation they might walk into is going to be running the latest version.
+- Some people need help with running the tests.
+- Check indentation.
+- Printing vs. returning.
```

Comment:
printing or output?  I don't remember the terms in Perl  (active circa 2000, so been a while) but output can go to standard error or standard out or another file, or of course it can go to a printer.  So should we use "print" or should we use "output" here?

```suggestion
- Output vs. return.
```

Reply:
people still use print, but you redirect it via &2> from the cli or you pick STDERR and friends. Defnitely stick with `print`, imo. Although, perl being perl, it probably doesn't make terribly much difference! Nobody knows what anyone means!

Reply:
That is true. lol

The return from the function and the output from that function may be two different things, and "printing" is a third thing.  So I want to make sure that it is clear what is being said, because learning is hard when you have to make the distinction of "What do they really mean with this word?  Are they speaking strictly, or in the more general sense they really mean the other thing this is commonly used for?"